### PR TITLE
Feature/s3storage

### PR DESCRIPTION
**Pull request by jnation3406**
This pull request adds in django-storages to store static and media files in an S3 bucket. 

By default (no environment variables set), if DEBUG is set to True, static files will be served through django's normal debug means - this can be done on the dev server or locally for testing. 

The production deploy will have the proper environment variables set to use the S3 bucket named 'observe.lco.global' to store static and media files. Each will be stored within it's own subdirectory in the bucket. Static files are publicly accessible in the bucket, but media files are private. 

A Jenkinsfile has been added to the project, which builds a container and runs unit tests for all commits, and for tagged commits to master it pushes the docker container and uploads the static files to S3. The developer should then update the container version in rancher to the newly tagged version as soon as possible. When we switch to kubernetes, we should be able to automate the deployment of the containers as well.

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <!doctype html>
-{% load static bootstrap3 %}
+{% load static from staticfiles %}
```
Comment (Fingel) [inline templates/base.html]: What's the difference between load static and load static from staticfiles?

File: valhalla/sciapplications/urls.py [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
-    SciApplicationDetailView, SciApplicationDeleteView, SciApplicationPDFView
+    SciApplicationDetailView, SciApplicationDeleteView, SciApplicationPDFView, SciApplicationGetPDFView
 )
 
 app_name = 'sciapplications'
 urlpatterns = [
     url(r'^$', SciApplicationIndexView.as_view(), name='index'),
     url(r'^(?P<pk>\d+)/$', SciApplicationDetailView.as_view(), name='detail'),
-    url(r'^(?P<pk>\d+)/pdf/$', SciApplicationPDFView.as_view(), name='pdf'),
+    url(r'^(?P<pk>\d+)/combined_pdf/$', SciApplicationPDFView.as_view(), name='pdf'),
```
Comment (Fingel) [inline valhalla/sciapplications/urls.py]: What is the second pdf view for?

File: valhalla/sciapplications/views.py [hunk 3]
```diff
@@ -216,6 +216,30 @@ def render_to_response(self, context, **kwargs):
         return pdf_response
 
 
+class SciApplicationGetPDFView(LoginRequiredMixin, DetailView):
```
Comment (Fingel) [inline valhalla/sciapplications/views.py]: This is nearly verbatim copy and paste from the original PDF view, but there is no explanation of what it is for or what makes it different.

File: templates/base.html (hunk 1 above)
Comment (Fingel) [reply templates/base.html]: https://docs.djangoproject.com/en/1.10/ref/templates/builtins/#static
Changed in Django 1.10:

In older versions, you had to use {% load static from staticfiles %} in your template to serve files from the storage defined in STATICFILES_STORAGE. This is no longer required.

Comment (jnation3406) [reply templates/base.html]: good catch

**Pull request merged**